R/cal-validate.R

```diff
@@ -0,0 +1,239 @@
+                                       summarize = TRUE,
+                                       ...) {
+  cal_validate(
+    rset = .data,
+    truth = {{ truth }},
+    estimate = {{ estimate }},
+    cal_function = "isotonic",
+    metrics = metrics,
+    summarize = summarize,
+    ...
+  )
+}
+
+#' @rdname cal_validate_logistic
+#' @inheritParams cal_estimate_beta
+#' @export
+cal_validate_beta <- function(.data,
+                              truth = NULL,
+                              shape_params = 2,
+                              location_params = 1,
+                              estimate = dplyr::starts_with(".pred_"),
+                              parameters = NULL,
+                              metrics = NULL,
+                              summarize = TRUE,
+                              ...) {
+  UseMethod("cal_validate_beta")
+}
+
+#' @export
+cal_validate_beta.rset <- function(.data,
+                                   truth = NULL,
+                                   shape_params = 2,
+                                   location_params = 1,
+                                   estimate = dplyr::starts_with(".pred_"),
+                                   parameters = NULL,
+                                   metrics = NULL,
+                                   summarize = TRUE,
+                                   ...) {
+  cal_validate(
+    rset = .data,
+    truth = {{ truth }},
+    estimate = {{ estimate }},
+    cal_function = "beta",
+    metrics = metrics,
+    summarize = summarize,
+    shape_params = shape_params,
+    location_params = location_params,
+    ...
+  )
+}
+
+cal_validate <- function(rset,
+                         truth = NULL,
+                         estimate = NULL,
+                         cal_function = NULL,
+                         metrics = NULL,
+                         summarize = TRUE,
+                         ...) {
+  if (is.null(cal_function)) rlang::abort("No calibration function provided")
+
+  if (is.null(metrics)) {
+    metrics <- yardstick::metric_set(
+      yardstick::brier_class
+    )
+  }
+
+  data_tr <- purrr::map(rset$splits, rsample::training)
+  data_as <- purrr::map(rset$splits, rsample::assessment)
+
+  if (cal_function == "logistic") {
+    cals <- purrr::map(
+      data_tr,
+      cal_estimate_logistic,
+      truth = {{ truth }},
+      estimate = {{ estimate }},
+      ...
+    )
+  }
+
+  if (cal_function == "isotonic") {
+    cals <- purrr::map(
+      data_tr,
+      cal_estimate_isotonic,
+      truth = {{ truth }},
+      estimate = {{ estimate }},
+      ...
+    )
+  }
+
+  if (cal_function == "beta") {
+    cals <- purrr::map(
+      data_tr,
+      cal_estimate_beta,
+      truth = {{ truth }},
+      estimate = {{ estimate }},
+      ...
+    )
+  }
+
+  estimate_col <- cals[[1]]$levels[[1]]
+
+  applied <- seq_along(data_as) %>%
+    map(
+      ~ {
+        val <- cal_apply(data_as[[.x]], cals[[.x]])
```

Comment:
Do we get the new results for `.pred_class`? I don't see that in the `validation` column when `summarize = FALSE`.

Reply:
```
res <- rsample::bootstraps(segment_logistic)

x <- cal_validate_beta(res, Class, summarize = FALSE, save_details = TRUE)

x$validation[[1]]
#> # A tibble: 381 × 4
#>    .pred_poor .pred_good Class .pred_class
#>         <dbl>      <dbl> <fct> <fct>      
#>  1      0.920   0.0800   poor  poor       
#>  2      0.244   0.756    good  good       
#>  3      0.235   0.765    good  good       
#>  4      0.998   0.00199  poor  poor       
#>  5      0.764   0.236    good  poor       
#>  6      0.999   0.000680 poor  poor       
#>  7      0.758   0.242    poor  poor       
#>  8      0.993   0.00650  poor  poor       
#>  9      0.272   0.728    poor  good       
#> 10      0.323   0.677    good  good       
#> # … with 371 more rows
```